app/pages/lab/workflow.jsx

```diff
@@ -0,0 +1,1042 @@
+    const disabledIfWorkflowInactive = classnames({ 'disabled-section': projectLiveWorkflowInactive });
+    const taskEditorClasses = classnames({'column': true, 'disabled-section': projectLiveWorkflowActive});
+
+    return (
+      <div className="edit-workflow-page">
+        <h3>{this.props.workflow.display_name} #{this.props.workflow.id}{' '}
+          <button onClick={this.showCreateWorkflow.bind(this)} disabled={this.state.workflowCreationInProgress} title="Copy workflow">
+            <i className="fa fa-copy"/>
+          </button>
+        </h3>
+        {this.state.workflowCreationInProgress ?
+          <ModalFormDialog tag="div">
+            <WorkflowCreateForm onSubmit={this.props.workflowActions.copyWorkflowForProject} onCancel={this.hideCreateWorkflow.bind(this)} onSuccess={this.handleWorkflowCreation.bind(this)}  project={this.props.project} workflowToClone={this.props.workflow} workflowActiveStatus={!this.props.project.live} />
+          </ModalFormDialog> : undefined}
+        <p className="form-help">A workflow is the sequence of tasks that you’re asking volunteers to perform. For example, you might want to ask volunteers to answer questions about your images, or to mark features in your images, or both.</p>
+        {this.props.project.live && this.props.workflow.active ?
+          <p className="form-help warning">
+            <strong>
+              You cannot edit tasks for an active workflow on a live project, all the task fields below will be inactive.
+              <br />
+              To edit these fields, deactivate your workflow on the <Link to={`/lab/${this.props.project.id}/workflows`}>workflows page</Link> - do not change your project to development.
+            </strong>
+          </p> : undefined}
+        <div className="columns-container">
+          <div className="column">
+            <div>
+              <AutoSave tag="label" resource={this.props.workflow}>
+                <label className="form-label" for="displayName">Workflow title</label>
+                <br />
+                <input type="text" id="displayName" name="display_name" value={this.props.workflow.display_name} className="standard-input full" onChange={handleInputChange.bind(this.props.workflow)} />
+              </AutoSave>
+              <small className="form-help">If you let your volunteers choose which workflow to attempt, this text will appear as an option on the project front page.</small>
+
+              <br />
+
+              <div className={disabledIfLive}>
+                <div className="nav-list standalone">
+                  <span className="nav-list-header">Tasks</span>
+                  <br />
+                  {noTasksDefined ?
+                    <div className="nav-list-item">
+                      <small className="form-help">(No tasks yet)</small>
+                    </div>
+                  :
+                    (() => {
+                      const result = [];
```

Comment:
Why is this declared as an anonymous function that runs automatically? It’s very hard to read.

Reply:
This is decaffeinate’s answer to mapping arrays.